main.py

```diff
@@ -292,6 +293,30 @@ def disable_switch(self, request: Request) -> JSONResponse:
         self.notify_topology_update()
         self.notify_switch_links_status(switch, "link disabled")
         return JSONResponse("Operation successful", status_code=201)
+    
+    @rest('v3/switches/{dpid}', methods=['DELETE'])
+    def delete_switch(self, request: Request) -> JSONResponse:
+        """Delete switch"""
+        dpid = request.path_params["dpid"]
+        with self._switch_lock:
+            try:
+                switch: Switch = self.controller.switches[dpid]
+            except KeyError:
+                raise HTTPException(404, detail="Switch not found.")
+            if switch.status != EntityStatus.DISABLED:
+                raise HTTPException(409, detail=f"The switch is not disable.")
+            links = self.links.copy()
+            for link_id, link in links.items():
+                if link.endpoint_a.switch.dpid == dpid or link.endpoint_b.switch.dpid == dpid:
+                    raise HTTPException(409, detail=f"This switch have link {link_id}")
```

Comment:
This makes a lot of sense, but let's instruct what the user needs to do too. You're trying to enforce that links should've been disabled, so ideally on this error message you probably want to instruct users to first disable of_lldp and delete links, otherwise links would keep getting rediscovered, and users might struggle to understand what they need to do.

Reply:
Changed message